rumqttd/src/link/remote.rs

```diff
@@ -225,3 +210,157 @@ where
     // Ok((connect, props, lastwill, lastwill_props))
     Ok(packet)
 }
+
+fn handle_auth(
+    config: Arc<ConnectionSettings>,
+    login: Option<&Login>,
+    client_id: String,
+) -> Result<(), Error> {
+    if config.static_auth.is_none() && config.dynamic_auth.is_none() {
+        return Ok(());
+    }
+
+    if login.is_none() && (config.static_auth.is_some() || config.dynamic_auth.is_some()) {
```

Comment:
we already checked that either of auth or dymanic auth is present ( i.e. both aren't None ). so no need to check again!
```suggestion
    if login.is_none()  {
```
PS: please add the comment why login being absent isn't allowed here ( same comments which existed before will work as well :) )